src/services/exit-logs/service.ts

```diff
@@ -0,0 +1,82 @@
+import { makeLogger } from '../../lib/index.js'
+
+import { ConfigService } from 'services/config/service.js'
+import { MetricsService } from '../prom/service'
+
+import { makeVerifier } from './verifier.js'
+import { ExecutionApiService } from '../execution-api/service.js'
+import { makeExitLogsFetcherService } from './fetcher.js'
+import { makeExitLogsCacheService } from './cache.js'
+
+export type ExitLogsService = ReturnType<typeof makeExitLogsService>
+
+const LOAD_LOGS_STEP = 10000
+
+export const makeExitLogsService = (
+  logger: ReturnType<typeof makeLogger>,
+  el: ExecutionApiService,
+  {
+    STAKING_MODULE_ID,
+    ORACLE_ADDRESSES_ALLOWLIST,
+    DISABLE_SECURITY_DONT_USE_IN_PRODUCTION,
+    BLOCKS_PRELOAD,
+  }: ConfigService,
+  metrics: MetricsService
+) => {
+  const verifier = makeVerifier(logger, el, {
+    STAKING_MODULE_ID,
+    ORACLE_ADDRESSES_ALLOWLIST,
+  })
+
+  const fetcher = makeExitLogsFetcherService(
+    logger,
+    verifier,
+    el,
+    {
+      STAKING_MODULE_ID,
+      DISABLE_SECURITY_DONT_USE_IN_PRODUCTION,
+    },
+    metrics
+  )
+
+  const cache = makeExitLogsCacheService()
+
+  const getLogs = async (operatorIds: number[], lastBlockNumber: number) => {
+    const header = cache.getHeader()
+
+    const initialBlockFrom = Math.max(0, lastBlockNumber - BLOCKS_PRELOAD)
+
+    // If data is already fully cached up to the latest block, return cached data
+    if (header.endBlock && header.endBlock === lastBlockNumber) {
```

Comment:
As I understand `header.endBlock` on init will be always 0, so we can simplify

```suggestion
    if (header.endBlock === lastBlockNumber) {
```